src/cluster.c

```diff
@@ -1301,52 +1359,32 @@ void valkeyClusterFree(valkeyClusterContext *cc) {
     vk_free(cc);
 }
 
-static valkeyClusterContext *
-valkeyClusterConnectInternal(valkeyClusterContext *cc, const char *addrs) {
-    if (valkeyClusterSetOptionAddNodes(cc, addrs) != VALKEY_OK) {
-        return cc;
-    }
-    valkeyClusterUpdateSlotmap(cc);
-    return cc;
-}
-
-valkeyClusterContext *valkeyClusterConnect(const char *addrs, int flags) {
-    valkeyClusterContext *cc;
-
-    cc = valkeyClusterContextInit();
-
+valkeyClusterContext *valkeyClusterConnectWithOptions(const valkeyClusterOptions *options) {
+    valkeyClusterContext *cc = valkeyClusterContextInit(options);
     if (cc == NULL) {
         return NULL;
     }
+    /* Only connect if options are ok. */
+    if (cc->err == 0) {
+        valkeyClusterUpdateSlotmap(cc);
+    }
+    return cc;
+}
 
-    cc->flags = flags;
+valkeyClusterContext *valkeyClusterConnect(const char *addrs) {
+    valkeyClusterOptions options = {0};
+    options.initial_nodes = addrs;
 
-    return valkeyClusterConnectInternal(cc, addrs);
+    return valkeyClusterConnectWithOptions(&options);
 }
 
 valkeyClusterContext *valkeyClusterConnectWithTimeout(const char *addrs,
```

Comment:
Is this kept for symmetry with valkeyConnectWithOptions? And for easier migration?

Reply:
Yes, it was kept for symmetry.
If we change the default timeouts to something reasonable the `valkeyClusterConnectWithTimeout` API is less useful and the `valkeyClusterConnect` API could easily be used instead when migrating.

Reply:
OK, we can still keep it for symmetry AND change the default timeout I guess.